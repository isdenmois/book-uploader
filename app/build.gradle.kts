plugins {
    alias(libs.plugins.application)
    alias(libs.plugins.kotlin.android)
    alias(libs.plugins.compose.compiler)
    alias(libs.plugins.ksp)
    alias(libs.plugins.hilt)
}

android {
    namespace = "com.isdenmois.bookuploader"

    compileSdk = Versions.sdk

    defaultConfig {
        applicationId = "com.isdenmois.bookuploader"
        minSdk = Versions.minSdk
        targetSdk = Versions.sdk
        versionCode = 3200
        versionName = "3.2.0"

        testInstrumentationRunner = "androidx.test.runner.AndroidJUnitRunner"

        vectorDrawables {
            useSupportLibrary = true
        }
    }

    buildTypes {
        named("release") {
            isMinifyEnabled = true
            isShrinkResources = true
            setProguardFiles(listOf(getDefaultProguardFile("proguard-android-optimize.txt"), "proguard-rules.pro"))
        }
    }
    compileOptions {
        sourceCompatibility = JavaVersion.VERSION_21
        targetCompatibility = JavaVersion.VERSION_21
    }
    kotlinOptions {
        jvmTarget = "21"
    }
    buildFeatures {
        compose = true
    }
    composeOptions {
        kotlinCompilerExtensionVersion = libs.versions.composeCompiler.get()
    }
    packaging {
        resources {
            excludes += arrayOf(
                "README.txt",
                "META-INF/*.md",
                "META-INF/LICENSE.md",
                "META-INF/LICENSE-notice.md",
                "**/attach_hotspot_windows.dll",
                "META-INF/AL2.0",
                "META-INF/LGPL2.1",
                "META-INF/licenses/ASM"
            )
        }
    }
}

dependencies {
    implementation(libs.core.ktx)
    implementation(platform(libs.compose.bom))
    implementation(libs.compose.ui)
    implementation(libs.compose.material)
    implementation(libs.compose.material.icons)
    implementation(libs.compose.ui.tooling.preview)
    implementation(libs.activity.compose)

    debugImplementation(libs.compose.ui.tooling)

    implementation(libs.ebookparser)

    // Coroutines
    implementation(libs.kotlinx.coroutines.core)
    implementation(libs.kotlinx.coroutines.android)

    // Hilt
    implementation(libs.hilt.android)
    ksp(libs.hilt.android.compiler)
    implementation(libs.lifecycle.viewmodel.compose)

    // Project dependencies
    implementation(project(ModuleDependency.core))
    implementation(project(ModuleDependency.domain))
    implementation(project(ModuleDependency.data))
    implementation(project(ModuleDependency.Feature.bookImport))
    implementation(project(ModuleDependency.Feature.profile))
    implementation(project(ModuleDependency.Feature.search))
    implementation(project(ModuleDependency.Feature.upload))

    // Testing
    testImplementation(libs.junit)
    androidTestImplementation(libs.junit.ext)
    androidTestImplementation(libs.espresso)
}
