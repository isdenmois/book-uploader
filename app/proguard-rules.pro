# This is a configuration file for R8
# keep everything in this package from being removed or renamed
-keep class com.isdenmois.** { *; }

# keep everything in this package from being renamed only
-keepnames class com.isdenmois.** { *; }
-allowaccessmodification
