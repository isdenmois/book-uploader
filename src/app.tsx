import React from 'react';
import { ActivityIndicator } from 'react-native';
import { NavigationContainer } from '@react-navigation/native';
import { createStackNavigator } from '@react-navigation/stack';
import { UploadScreen } from './pages/upload/upload.screen';
import { ScanScreen } from 'pages/scan';
import { HomeScreen } from 'pages/home';
import { AddressContext, useCreateAddressContext } from 'utils/address';

const Stack = createStackNavigator();

const App = () => {
  const context = useCreateAddressContext();

  if (context.address === null) {
    return <ActivityIndicator size='large' />;
  }

  return (
    <AddressContext.Provider value={context}>
      <NavigationContainer>
        <UploadScreen>
          <Stack.Navigator initialRouteName={context.address ? 'home' : 'scan'}>
            <Stack.Screen name='scan' component={ScanScreen} options={{ title: '' }} />
            <Stack.Screen name='home' component={HomeScreen} options={{ header: () => null }} />
          </Stack.Navigator>
        </UploadScreen>
      </NavigationContainer>
    </AddressContext.Provider>
  );
};

export default App;
