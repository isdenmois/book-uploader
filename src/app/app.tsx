import React, { useEffect } from 'react'
import { RecoilRoot } from 'recoil'
import { StatusBar, StatusBarStyle, useColorScheme } from 'react-native'
import { NavigationContainer } from '@react-navigation/native'
import { createStackNavigator } from '@react-navigation/stack'
import changeNavigationBarColor from 'react-native-navigation-bar-color'

import { ScanScreen } from 'screens/scan'
import { MainScreen } from 'screens/main'
import { ThemeProvider, useTheme } from 'shared/ui'
import { useMemo } from 'react'

const Stack = createStackNavigator()

const header = () => null

export const App = () => {
  return (
    <ThemeProvider>
      <RecoilRoot>
        <NavigationContainer>
          <StackNavigator />
        </NavigationContainer>
      </RecoilRoot>
    </ThemeProvider>
  )
}

function StackNavigator() {
  const mode = useColorScheme()
  const { colors } = useTheme()
  const barStyle: StatusBarStyle = mode === 'dark' ? 'light-content' : 'dark-content'

  const s = useMemo(
    () => ({
      cardStyle: {
        backgroundColor: colors.background,
      },
      modalCardStyle: {
        backgroundColor: 'transparent',
      },
    }),
    [mode],
  )

  useEffect(() => {
    changeNavigationBarColor(colors.tabsBackground, mode === 'light', false)
  }, [mode])

  return (
    <>
      <StatusBar backgroundColor={colors.background} barStyle={barStyle} />
      <Stack.Navigator screenOptions={{ cardStyle: s.cardStyle }}>
        <Stack.Screen name='main' component={MainScreen} options={{ header }} />
        <Stack.Screen
          name='scan'
          component={ScanScreen}
          options={{
            title: '',
            animationEnabled: false,
            header,
            cardStyle: s.modalCardStyle,
            cardOverlayEnabled: false,
          }}
        />
      </Stack.Navigator>
    </>
  )
}
