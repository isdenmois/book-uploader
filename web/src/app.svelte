<script>
  import Search from './search.svelte';
  import { searchHandler } from './api';

  let books = [];

  async function search({ detail: { query } }) {
    const { data, error } = await searchHandler(query);

    if (error) {
      return alert(error);
    }

    console.log(data);

    books = data;
  }
</script>

<main>
  <Search on:search={search} />

  <ul>
    {#each books as { link, title, authors }, i}
      <li><a target="_blank" href="/api/rewrite?path={link}"> {title} </a></li>
    {/each}
  </ul>
</main>
