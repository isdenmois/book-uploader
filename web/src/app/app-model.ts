import { computed } from 'nanostores'
import { $auth } from 'features/auth'
import { $source } from 'features/filters'

export const $showLogin = computed([$source, $auth], (source, cookie) => source === 'ZLIB' && !cookie)
