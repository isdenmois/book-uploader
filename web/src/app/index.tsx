import { render } from 'solid-js/web'
import 'virtual:windi.css'

import { App } from './app'

render(() => <App />, document.getElementById('app') as HTMLElement)
