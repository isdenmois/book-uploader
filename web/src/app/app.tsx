import type { Component } from 'solid-js'
import { Dynamic } from 'solid-js/web'

import { useSignal } from 'shared/lib/solid-nanostore'
import { SearchPage } from 'pages/search-page'
import { LoginPage } from 'pages/login-page'

import { $showLogin } from './app-model'

export const App: Component = () => {
  const showLogin = useSignal($showLogin)

  return <Dynamic component={showLogin() ? LoginPage : SearchPage} />
}
