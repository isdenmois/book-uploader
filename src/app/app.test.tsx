import { render } from '@testing-library/react-native'
import React from 'react'

import 'react-native-gesture-handler/jestSetup'
jest.mock('./navigation', () => ({ MainNavigator: () => null }))
jest.mock('@react-native-community/async-storage', () => ({ getItem: () => null }))
jest.mock('@react-navigation/native', () => ({
  NavigationContainer({ children, ...props }) {
    const Element: any = 'NavigationContainer'

    return <Element {...props}>{children}</Element>
  },
}))
jest.mock('react-native-fs', () => ({}))

import { App } from '../app'

test('App', () => {
  const { toJSON } = render(<App />)

  expect(toJSON()).toMatchSnapshot()
})
