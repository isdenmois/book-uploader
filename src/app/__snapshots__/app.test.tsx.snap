// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App 1`] = `
<NavigationContainer>
  <Navigator
    screenOptions={
      Object {
        "cardStyle": Object {
          "backgroundColor": "#F9FAFE",
        },
      }
    }
  >
    <Screen
      component={[Function]}
      name="main"
      options={
        Object {
          "header": [Function],
        }
      }
    />
    <Screen
      component={[Function]}
      name="scan"
      options={
        Object {
          "animationEnabled": false,
          "cardOverlayEnabled": false,
          "cardStyle": Object {
            "backgroundColor": "transparent",
          },
          "header": [Function],
          "title": "",
        }
      }
    />
  </Navigator>
</NavigationContainer>
`;
